src/middlewared/middlewared/plugins/gluster_linux/volume.py

```diff
@@ -0,0 +1,542 @@
+        # glustercli-python has a bug where if provided the "force"
+        # option, it will concatenate it with the "start" option
+        # This is wrong, you can choose "start" or "force" exclusively
+        # (i.e. gluster volume volname remove-brick peer:path start OR force)
+        if op.lower() == 'start':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_start, volname, bricks, **data)
+
+        if op.lower() == 'stop':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_stop, volname, bricks, **data)
+
+        if op.lower() == 'commit':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_commit, volname, bricks, **data)
+
+        if op.lower() == 'status':
+            result = self.__volume_wrapper(
+                volume.bricks.remove_status, volname, bricks, **data)
+
+        return result
+
+    @private
+    def replacebrick_volume(self, data):
+
+        volname = data.pop('volname')
+        src = data.pop('src_brick')
+        new = data.pop('new_brick')
+
+        src_peer = src['peer_name']
+        src_path = src['peer_path']
+        src_brick = src_peer + ':' + src_path
+
+        new_peer = new['peer_name']
+        new_path = new['peer_path']
+        new_brick = new_peer + ':' + new_path
+
+        result = self.__volume_wrapper(
+            volume.bricks.replace_commit,
+            volname,
+            src_brick,
+            new_brick,
+            **data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_create',
+        Str('volname', required=True),
+        List('volbricks', items=[
+            Dict(
+                'brick',
+                Str('peer_name', required=True),
+                Str('peer_path', required=True),
+            ),
+        ], required=True),
+        Int('replica'),
+        Int('stripe'),
+        Int('arbiter'),
+        Int('disperse'),
+        Int('disperse_data'),
+        Int('redundancy'),
+        Bool('force'),
+    ))
+    @job(lock='glustervolume_create')
+    def do_create(self, job, data):
+        """
+        Create a gluster volume.
+
+        `volname` Name to be given to the gluster volume
+        `volbricks` List of brick paths
+            `peer_name` IP or DNS name of the peer.
+            `peer_path` The full path of the brick
+
+        `replica` Number of replica bricks
+        `stripe` Number of stripe bricks
+        `arbiter` Number of arbiter bricks
+        `disperse` Number of disperse bricks
+        `disperse_data` Number of disperse data bricks
+        `redundancy` Number of redundancy bricks
+        `force` Create volume forcefully, ignoring potential warnings
+        """
+
+        self.common_validation(job)
+
+        return self.create_volume(data)
+
+    @accepts(Dict(
+        'glustervolume_start',
+        Str('volname', required=True),
+        Bool('force'),
+    ))
+    @job(lock='glustervolume_start')
```

Comment:
Is the volume name always the unique identifier for all operations related to volume?

If thats the case I think we should change all these methods to comply with other CRUD services we have, and so the REST API is based on that.

e.g.

@item_method
def start(self, job, id, options):
  ...

For anything other than create/update/delete that uses the volume name as unique identifier.

That way for REST it would also work as:
/gluster/volume/id/{name}/start/

I dont know if it works with a job (never tried that), needs verification.

Reply:
Yes, the volume name is always the unique identifier. I'll make this comply like other REST APIs.

Reply:
`@item_method` can be used with `@job` decorator. I confirmed testing it. I've pushed PR fixing this.